### PR TITLE
Do inventory date range validation on the backend

### PR DESCRIPTION
**Pull request by gigxz**
We soft-validate this on the frontend (aka show warnings if outside min/max), but we should validate it on the backend too.

We could also/instead make the frontend do a hard validation, and not allow submitting if date is outside of bounds. that would require some reconfiguring of form state (doable, just more effort than this).

### CODE REVIEW

File: drivers/hmis/spec/requests/hmis/update_inventory_spec.rb [hunk 1]
```diff
@@ -120,6 +122,58 @@
         expect(errors[0]['type']).to eq 'required'
       end
     end
+
+    it 'validates start date against project operating period' do
```
Comment (eanders) [inline drivers/hmis/spec/requests/hmis/update_inventory_spec.rb]: I think we should check that it passes validation if the inventory start/end are on the project start end.    I'm having trouble telling if that check is here.

**Review by eanders — APPROVED** [review]
Looks great!  But if there aren't positive and negative edge case tests, can you add them?

File: drivers/hmis/spec/requests/hmis/update_inventory_spec.rb (hunk 1 above)
Comment (gigxz) [reply drivers/hmis/spec/requests/hmis/update_inventory_spec.rb]: Thanks - I added a bunch more tests
